### PR TITLE
export `createEventHandlerWithConfig()` and add documentation.

### PR DESCRIPTION
**Pull request by acchou**
Address issue #520. I'll also suggest a separate PR for typescript types in the appropriate repo.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -889,13 +889,28 @@ const Timer = enhance(({ timeElapsed }) =>
...
-  handler: (value: T) => void
-  stream: Observable<T>,
+  handler: (value: T) => void,
+  stream: Observable<T>
 }
 ```
 
 Returns an object with properties `handler` and `stream`. `stream` is an observable sequence, and `handler` is a function that pushes new values onto the sequence. Useful for creating event handlers like `onClick`.
 
+### `createEventHandlerWithConfig()`
```
Comment (wuct) [inline docs/API.md]: Could you also add an entry to the TOC section?

Comment (wuct) [reply docs/API.md]: It turns out that `mapPropsStreamWithConfig` is also missing from the TOC section.

File: docs/API.md [hunk 2]
```diff
@@ -889,13 +889,28 @@ const Timer = enhance(({ timeElapsed }) =>
...
+```js
+createEventHandlerWithConfig<T>(
+  config: {
+    fromESObservable<T>: ?(observable: Observable<T>) => Stream<T>,
+    toESObservable<T>: ?(stream: Stream<T>) => Observable<T>,
+  }
+) => (): {
+  handler: (value: T) => void
```
Comment (wuct) [inline docs/API.md]: missing a `,` here.

File: docs/API.md (hunk 1 above)
Comment (acchou) [reply docs/API.md]: Added these functions to the TOC, and also `componentFromStreamWithConfig()`.

**Comment by acchou** [comment]
I noticed that I had to npm install `change-emitter` and `hoist-non-react-statics` in order to push. Should these deps be in `package.json`?

**Comment by wuct** [comment]
@acchou Thanks! And yes, we can put them in `devDependencies`.

**Pull request merged**